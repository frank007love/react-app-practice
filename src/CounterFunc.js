import React, { useState } from "react"

function CounterFunc() {
    const [ count, setCount ] = useState(0);

    return (
        <>
            <div>This is a example: {count}.</div>
            <button onClick={()=>setCount(count+1)}>count+1</button>
        </>
    )
}

export default CounterFunc