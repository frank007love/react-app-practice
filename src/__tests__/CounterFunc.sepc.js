import CounterFunc from "../CounterFunc";
import {render, fireEvent, cleanup} from 'react-testing-library';
import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';


describe('<CounterFunc>',()=>{
    afterEach(cleanup);

    it('CounterFunc snapshot', () => {
        const tree = renderer.create(<CounterFunc/>).toJSON()
        expect(tree).toMatchSnapshot();
    })

    // //Can't work with React-Hook
    // it('Test click button of counterFunc', ()=>{
    //     const counterFunc = shallow(<CounterFunc/>);
    //     counterFunc.find('button').simulate('click');
    //     expect(counterFunc.find('div').text()).toBe('This is a example: 1.');
    // });

    it('Test click button of counterFunc', ()=>{
        const utils = render(<CounterFunc/>);
        const display = utils.getByLabelText('counter-val');
        const button = utils.getByText('count+1');

        fireEvent.click(button);
        expect(display.textContent).toBe('This is a example: 1.');
    });
});