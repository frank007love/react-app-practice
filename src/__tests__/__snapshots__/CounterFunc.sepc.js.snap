// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterFunc> CounterFunc snapshot 1`] = `
Array [
  <div
    aria-label="counter-val"
  >
    This is a example: 
    0
    .
  </div>,
  <button
    onClick={[Function]}
  >
    count+1
  </button>,
]
`;
